site/journal/Ember Template Imports/Part 1.md

```diff
@@ -170,7 +170,7 @@ This example introduces a stateful, class-backed component which uses a bound fu
         {{on "input" this.updateName}}
       />
 
-      <button type='submit' disabled={{eq this.value.length 0}}>
+      <button type='submit' disabled={{eq this.name.length 0}}>
```

Comment:
BTW I use to use `{{not this.name}}`, which, given that `name` is string (or string | null | undefined) type, should be the same thing. Are you checking the length instead for clarity purposes?

Reply:
I opted not to use `not` for a couple reasons, yeah: first, as you say; and second, I try not to have `null | undefined` in my types *at all, ever* when possible. (Usually, though not in this example, I’d be using TypeScript to enforce that statically!)